app/src/main/java/com/android/solvit/shared/model/chat/ChatAssistantViewModel.kt

```diff
@@ -0,0 +1,157 @@
+            return ChatAssistantViewModel() as T
+          }
+        }
+  }
+
+  /**
+   * Set the context for the chat assistant to generate a response
+   *
+   * @param messageContext List of previous messages in the conversation
+   * @param senderName Name of the sender
+   * @param receiverName Name of the receiver
+   * @param requestContext (Optional) The concerned service request
+   */
+  fun setContext(
+      messageContext: List<ChatMessage.TextMessage>,
+      senderName: String,
+      receiverName: String,
+      requestContext: ServiceRequest? = null
+  ) {
+    _messageContext.value = messageContext
+    _senderName.value = senderName
+    _receiverName.value = receiverName
+    _requestContext.value = requestContext
+  }
+
+  /**
+   * Add a new message to the context
+   *
+   * @param message The new message to add
+   */
+  fun updateMessageContext(message: ChatMessage.TextMessage) {
+    _messageContext.value += message
+  }
+
+  /**
+   * Update the selected tones of the response
+   *
+   * @param tones List of selected tones
+   */
+  fun updateSelectedTones(tones: List<String>) {
+    _selectedTones.value = tones
+  }
+
+  /** Clear all the context values */
+  fun clear() {
+    _messageContext.value = emptyList()
+    _senderName.value = ""
+    _receiverName.value = ""
+    _requestContext.value = null
+    _selectedTones.value = emptyList()
+  }
+
+  /**
+   * Build the prompt for the chat assistant
+   *
+   * @param input Additional infos to add to the prompt
+   * @return The generated prompt
+   */
+  fun buildPrompt(input: String): String {
+    var prompt = "Write a single message response"
+    if (senderName.value.isNotEmpty()) {
+      prompt += " from " + senderName.value
+    }
+    if (receiverName.value.isNotEmpty()) {
+      prompt += " to " + receiverName.value
+    }
+    if (_messageContext.value.isNotEmpty()) {
+      prompt += ", based on the following conversation:\n"
+      prompt += messageContext.value.joinToString("\n") { it.senderName + ": " + it.message }
+    }
+    if (_requestContext.value != null) {
+      prompt += ", based on the following service request:\n"
+      prompt += requestContext.value!!.title + ": " + requestContext.value!!.description
+    }
+    if (_selectedTones.value.isNotEmpty()) {
+      prompt += ", with the following tones:\n"
+      prompt += selectedTones.value.joinToString(", ")
+    }
+    if (input.isNotEmpty()) {
+      prompt += ", with the following infos provided by the sender:\n"
+      prompt += input
+    }
+    return prompt
+  }
+
+  /**
+   * Generate a response message from the chat assistant
+   *
+   * @param input Additional infos to add to the prompt
+   * @param onResponse Callback to handle the response
+   */
+  fun generateMessage(input: String, onResponse: (String) -> Unit) {
+    val prompt = buildPrompt(input)
+    Log.d("ChatAssistantViewModel", "Prompt: $prompt")
```

Comment:
Log.d call in generateMessage logs the entire prompt. I'm asking myself if we have to preserve a kind of confidentiality or not ! We can ask the coach about it new meeting !

Reply:
I can remove it!